src/modal/modal.scss

```diff
@@ -11,15 +11,53 @@
   left: 0;
   width: 100%;
   height: 100%;
+  @include themed {
+    background-color: rgba(0, 0, 0, t(iui-opacity-4));
+  }
+
+  // #region Small Modal Animations for CSS
   visibility: hidden;
   opacity: 0;
   @media (prefers-reduced-motion: no-preference) {
```

Comment:
It should be enough just to have transition in one place, you don't need to duplicate in every class.